### ISSUE TITLE
code 400, message Bad request version

### ISSUE THREAD

**Issue by j3mu5** [felixschndr/mealie-bring-api#4]
Hello, the container works fine, the connection via the recipe action works (URL is `https://XX.XX.XX.XX:YYYY/` since http does not lead to any results in the log). However, no recipes are being added to Bring, as the log shows:

```txt
[2024-07-19 19:52:08,938] [BringHandler] [INFO] [Found the bring list Einkauf (UUID: xxxx)]
[2024-07-19 19:52:08,508] [BringHandler] [INFO] [Login successful]
[2024-07-19 19:54:23,606] [werkzeug] [ERROR] [XX.XX.XX.XX - - [19/Jul/2024 19:54:23] code 400, message Bad request version ('\x02h2\x08http/1.1ÿ\x01\x00\x01\x00\x00#\x00\x00\x00')]
[2024-07-19 19:54:23,600] [werkzeug] [ERROR] [XX.XX.XX.XX - - [19/Jul/2024 19:54:23] code 400, message Bad HTTP/0.9 request type ('\x16\x03\x01\x06À\x01\x00\x06¼\x03\x03\x9eê\x15.\x11rU49¶\x97ðÕ\x02\x90AÄ\x9b')]
```

Edit: `curl -I http://XX.XX.XX.XX:YYYY` results in

```
HTTP/1.1 405 METHOD NOT ALLOWED
Server: Werkzeug/3.0.3 Python/3.10.14
Date: Fri, 19 Jul 2024 20:04:28 GMT
Content-Type: text/html; charset=utf-8
Allow: OPTIONS, POST
Content-Length: 153
Connection: close
```

**Comment by felixschndr** [comment]
The problem is that you are doing an HTTPS request (which is encrypted) and the server cannot decrypt it.

Why does HTTP not work?

**Comment by felixschndr** [comment]
> Thanks @felixschndr for the tip! http is the way to go.

👍🏻
-----

> However, when I add the IP addresses with port (http://XX.XX.XX.XX:YYYY/ including the / ) in Mealie in the recipe actions, I see no reaction to the recipe action your container (and also no recipes in my bring list).

This sounds like that the container can't reach the API to me. When you did `curl -I http://XX.XX.XX.XX:YYYY/status` I assume you did this _anywhere_ but not in the mealie container, right? 

Can you please go into the mealie container and do the curl again and see what happens?
```bash
docker exec -it -u root mealie_app /bin/bash # Go into the container
apt update && apt install curl -y # Install curl
curl -I http://XX.XX.XX.XX:YYYY/status
```
This _should_ also return an `OK`, but I assume it won't.

**Comment by felixschndr** [comment]
Let me rephrase that to see if I understand you correctly?
You are using Nginx as a reverse proxy (so do I) and access Mealie through it. In this case sending the request to the API does not work.
When you are using the direct access via IP and port (not over Nginx) the request works.

1. Is this correct?
2. I guess this has something to do with CORS policies. I don't know much about this but I think your browser has some kind of problem sending the request to the API while you are accessing Mealie via HTTPS.
3. Does the browser trust your self signed cert? (I can always recommend to buy a cheap domain and get a LetsEncrpyt cert which the browser will trust, this is what I do)
4. Do you have your Mealie Bring API behind the Nginx as well? As I see it you don't, right? Please try to add the API to a proxy host as well such that it is reachable via HTTP*S* and see what happens

**Comment by felixschndr** [comment]
> Problem was me not using http/https correctly in nginx settings.

Yeah, this `('\x16\x03\x01\x00·\x01\x00\x00³\x03\x03·ï\x9c\x08\x18`')]` is casual HTTPS not being decrypted. Glad it works now!

**Comment by hyzteric** [comment]
So for those using NPM (Nginx Proxy Manager), this is the setting that you have to change. Use the https on the external, but the internal should point to http://yourip:yourport/ like this

<img width="489" height="521" alt="Image" src="https://github.com/user-attachments/assets/7375d4d5-7e3f-4d40-91d0-d6ef529dd437" />

<img width="485" height="386" alt="Image" src="https://github.com/user-attachments/assets/4f888a08-d62a-4224-9bd7-23ce61224d25" />
